gadget/layout.go

```diff
@@ -310,6 +314,11 @@ func resolveVolumeContent(gadgetRootDir, kernelRootDir string, kernelInfo *kerne
 // content and populates VolumeContent.resolvedSource with absolute
 // paths.
 func resolveContentPathOrRef(gadgetRootDir, kernelRootDir string, kernelInfo *kernel.Info, pathOrRef string) (string, error) {
+	// TODO: what to here? we could error instead?
+	if pathOrRef == "" {
```

Comment:
to be clear, this error is when you have a nonsensical content specification like this in your gadget.yaml:

```yaml
    structure:
      - name: bleh
        type: bleh
        content:
          - image: ""
```

or does Content.UnresolvedSource get verified somewhere else first? 

If Content.UnresolvedSource does not get verified anywhere else I think this should be an error here

Reply:
ideally that should be verified earlier, and produce an internal error here?

Reply:
Makes sense, I will update in a followup